util/src/builder/callback_data.rs

```diff
@@ -0,0 +1,175 @@
+use twilight_model::{
+    application::{callback::CallbackData, component::Component},
+    channel::{
+        embed::Embed,
+        message::{AllowedMentions, MessageFlags},
+    },
+};
+
+/// Builder for [`CallbackData`].
+///
+/// # Example
+/// ```
+/// use twilight_util::builder::CallbackDataBuilder;
+/// use twilight_model::channel::message::MessageFlags;
+///
+/// let callback_data = CallbackDataBuilder::new()
+///     .content("My callback message")
+///     .flags(MessageFlags::EPHEMERAL)
+///     .build();
+/// ```
+#[derive(Debug)]
+#[must_use = "builders have no effect if unused"]
+pub struct CallbackDataBuilder(CallbackData);
+
+impl CallbackDataBuilder {
+    /// Create a new builder to construct a [`CallbackData`].
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Set allowed mentions in the interaction response.
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn allowed_mentions(mut self, allowed_mentions: AllowedMentions) -> Self {
+        self.0.allowed_mentions = Some(allowed_mentions);
+
+        self
+    }
+
+    /// Add a message component.
+    ///
+    /// Multiple component can be set by calling this method multiple times.
+    pub fn component(mut self, component: Component) -> Self {
```

Comment:
for this method, i would prefer a signature of `impl IntoIterator<Item = Component>` that replaces the list of components